test/new-e2e/tests/agent-runtimes/check_disk/disk_test.go

```diff
@@ -0,0 +1,167 @@
+		name        string
+		checkConfig string
+		agentConfig string
+	}{
+		{
+			"default",
+			`init_config:
+instances:
+  - use_mount: false
+`,
+			``,
+		},
+	}
+
+	for _, testCase := range testCases {
+		v.Run(testCase.name, func() {
+			v.T().Log("run the disk check using old version")
+			pythonMetrics := v.runDiskCheck(testCase.agentConfig, testCase.checkConfig, false)
+			v.T().Log("run the disk check using new version")
+			goMetrics := v.runDiskCheck(testCase.agentConfig, testCase.checkConfig, true)
+
+			// assert the check output
+			elementsMatch(v.T(), pythonMetrics, goMetrics, metricPayloadCompare)
+			v.T().Logf("the check emitted %d metrics", len(pythonMetrics))
+		})
+	}
+}
+
+func metricPayloadCompare(a, b check.Metric) bool {
+	return a.Host == b.Host &&
+		a.Interval == b.Interval &&
+		a.Metric == b.Metric &&
+		slices.EqualFunc(a.Points, b.Points, func(a, b []float64) bool {
+			return slices.EqualFunc(a, b, func(a, b float64) bool {
+				if a == 0 {
+					return b == 0
+				}
+				return math.Abs(a-b)/a <= metricCompareTolerance
+			})
+		}) &&
+		a.SourceTypeName == b.SourceTypeName &&
+		a.Type == b.Type &&
+		compareTags(a.Tags, b.Tags)
+}
+
+// compare the tags in both slices
+// not sure if there can be duplicates in the tags, but let's assume so
+// this is inefficient, but good enough for this test
+func compareTags(a, b []string) bool {
+	if len(a) != len(b) {
+		return false
+	}
+	tagsA := make(map[string]int, len(a))
+	for _, tag := range a {
+		tagsA[tag]++
+	}
+	for _, tag := range b {
+		tagsA[tag]--
+	}
+	for _, count := range tagsA {
+		if count != 0 {
+			return false
+		}
+	}
+	return true
+}
+
+func (v *baseCheckSuite) runDiskCheck(agentConfig string, checkConfig string, useNewVersion bool) []check.Metric {
+	v.T().Helper()
+
+	diskCheckVersion := "old"
+	if useNewVersion {
+		agentConfig += "\nuse_diskv2_check: true"
+		checkConfig += "\n    loader: core"
+		diskCheckVersion = "new"
+	}
+	diskCheckVersionTag := fmt.Sprintf("disk_check_version:%s", diskCheckVersion)
+	checkConfig += fmt.Sprintf("\n    tags:\n      - %s", diskCheckVersionTag)
+
+	v.UpdateEnv(awshost.Provisioner(
+		awshost.WithAgentOptions(
+			agentparams.WithAgentConfig(agentConfig),
+			agentparams.WithIntegration("disk.d", checkConfig),
+		),
+	))
+
+	// run the check
+	output := v.Env().Agent.Client.Check(agentclient.WithArgs([]string{"disk", "--json"}))
+	data := check.ParseCheckOutput(v.T(), []byte(output))
+
+	require.Len(v.T(), data, 1)
+	metrics := data[0].Aggregator.Metrics
+	for i := range metrics {
+		// remove the disk_check_version tag
+		tagLen := len(metrics[i].Tags)
+		metrics[i].Tags = slices.DeleteFunc(metrics[i].Tags, func(tag string) bool {
+			return tag == diskCheckVersionTag
+		})
+		if !assert.Equalf(v.T(), 1, tagLen-len(metrics[i].Tags), "expected tag %s once in metric %s, but found %d times", diskCheckVersion, metrics[i].Metric, tagLen-len(metrics[i].Tags)) {
+			v.T().Logf("metric: %+v", metrics[i])
+		}
+	}
```

Comment:
I looks a bit complex the check for the custom tag 😄 

```go
for i := range metrics {
	tagsWithoutCustomTags := []string{}
	foundCustomTags := false
	// remove the disk_check_version tag
	for _, t := range metrics[i].Tags {
		if t == diskCheckVersionTag {
			foundCustomTags = true
			continue
		}
		tagsWithoutCustomTags = append(tagsWithoutCustomTags, t)
	}

	if !foundCustomTags {
		v.T().Logf("metric %s does not have the disk_check_version tag", metrics[i].Metric)
	}
	metrics[i].Tags = tagsWithoutCustomTags
}
```

Do this alternatives do the same thing?

Reply:
This wouldn't detect if it's twice in the list I believe ?

Do you really find the code complex though ? It just removes the tag and asserts only one element was removed
I can store `tagLen-len(metrics[i].Tags)` in a variable to make it clear it's the number of elements removed if you want (and maybe shorten the assert message)

Reply:
I updated the code a bit, let me know if it's better

Reply:
> This wouldn't detect if it's twice in the list I believe ?

You are right about that 😄

The updated code looks much readable 👏